src/base/bittorrent/session.cpp

```diff
@@ -321,6 +321,11 @@ QString Session::tempPath() const
     return m_tempPath;
 }
 
+QString Session::torrentTempPath(const InfoHash &hash) const
+{
+    return m_tempPath + QString(hash) + "/";
```

Comment:
We can do, for example, as Git does - only use the first 7 characters of hash. This, of course, will spoil the uniqueness of the names, but collisions will still be unlikely (what are the odds that two torrent have matching first 7 characters of the hash and the same file names?).